src/midi/cake/tree_serializer.rs

```diff
@@ -66,11 +67,8 @@ impl TreeSerializer {
                 Some(pos) => -pos,
                 None => {
                     let written_pos = self.written_values.len() as i32;
-                    self.written_values.push(IntVector4::new_note(
-                        marker.start,
-                        0,
-                        marker.track_channel,
-                    ));
+                    self.written_values
+                        .push(IntVector4::new_note(marker.start, 0, marker.color));
```

Comment:
I'm honestly a little confused with this, the purpose of track_channel is to identify the color index in the shader rather than storing the color itself. What approach are you trying to do?

Reply:
The color was calculated inside that `new_note` function at first and it didn't account for the settings (it was hardcoded to rainbow). So I made the color vector be calculated when the parse function is called, where we know the palette settings (random/rainbow), and such the color of each note is passed as the argument instead of the track_channel number.

Reply:
ohh right I see what I did
anyway lgtm then

Reply:
Oh wait then I guess I could completely remove the `track_channel` variable because it's not used somewhere else. Will do that and merge

Reply:
Nvm it is used, merging as is